### PR TITLE
fix: Upgrading import paths for casbin/v2 and fixing travis

### PR DESCRIPTION
**Pull request by hiteshsethi**
This PR fixes:
1. import paths relative to casbin/v2
2. Introduces go mod in this pkg
3. Fixes travis-ci issues

### CODE REVIEW

**Comment by hiteshsethi** [comment]
cc: @hsluoyz

File: .travis.yml [hunk 1]
```diff
@@ -10,10 +13,8 @@ service:
...
   - echo $TRAVIS_GO_VERSION
-  - go get github.com/mattn/goveralls
-  - go get go.etcd.io/etcd/clientv3
   - docker pull quay.io/coreos/etcd:v3.3.10
   - docker run -d -p 2379:2379 quay.io/coreos/etcd:v3.3.10 etcd --name casbin --listen-client-urls http://0.0.0.0:2379 --advertise-client-urls http://0.0.0.0:2379
 
 script:
-  - $HOME/gopath/bin/goveralls -service=travis-ci
\ No newline at end of file
+  - make test
```
Comment (hsluoyz) [inline .travis.yml:20]: Why remove coveralls?

File: .travis.yml [hunk 2]
```diff
@@ -2,6 +2,9 @@ language: go
 
 sudo: required
 
+env:
+  - GO111MODULE=on
```
Comment (hsluoyz) [inline .travis.yml:6]: Do we need this for recent Go versions?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+default: lint test
```
Comment (hsluoyz) [inline Makefile:1]: Why introduce another makefile?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: Makefile (hunk 3 above)
Comment (hiteshsethi) [reply Makefile:1]: Currently etcd-watcher doesn't have any Makefile
I kinda checked the reference of this from xorm-adapter repository

Comment (closetool) [reply Makefile:1]: @hiteshsethi I suggest to remove this.

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,20 @@
+module github.com/casbin/etcd-watcher
+
+go 1.13
```
Comment (closetool) [inline go.mod:3]: plz use 1.14

**Review by closetool — CHANGES_REQUESTED** [review]
could you change travis into github action, like https://github.com/casbin/gorm-adapter/blob/master/.github/workflows/ci.yml

File: Makefile (hunk 3 above)
Comment (hsluoyz) [reply Makefile:1]: Same feeling

**Comment by hsluoyz** [comment]
@closetool we can switch to GitHub Actions in next PR. @hiteshsethi

File: .gitignore [hunk 5]
```diff
@@ -14,4 +14,7 @@
 .glide/
 
 .idea/
-*.iml
\ No newline at end of file
+*.iml
+
+#vendor files
+vendor
```
Comment (hsluoyz) [inline .gitignore:20]: Remove this change, we don't use `vendor` ever.

File: .travis.yml [hunk 6]
```diff
@@ -3,6 +3,7 @@ language: go
 sudo: required
 
 go:
+  - "1.14"
```
Comment (hsluoyz) [inline .travis.yml:6]: Remove this change. 1.14 is too too old.

File: go.mod [hunk 7]
```diff
@@ -0,0 +1,20 @@
+module github.com/casbin/etcd-watcher
+
+go 1.14
```
Comment (hsluoyz) [inline go.mod:3]: No 1.14

**Review by hsluoyz — CHANGES_REQUESTED** [review]

**Comment by hsluoyz** [comment]
@hiteshsethi I suggest installing latest Go 1.16.x and then do the PR again.

**Comment by hiteshsethi** [comment]
review comments are fixed
cc: @hsluoyz

**Comment by hiteshsethi** [comment]
> @hiteshsethi I suggest installing latest Go 1.16.x and then do the PR again.

I've been using go1.15.6 
Any reason for asking?

**Comment by hsluoyz** [comment]
For:

image

**Comment by hiteshsethi** [comment]
Did this because of 
image

**Comment by hsluoyz** [comment]
I suggesting using latest.

File: .travis.yml (hunk 6 above)
Comment (hiteshsethi) [reply .travis.yml:6]: Do you want to consider writing 1.16 here?

File: go.mod (hunk 7 above)
Comment (hiteshsethi) [reply go.mod:3]: updating to 1.16

File: .gitignore (hunk 5 above)
Comment (hiteshsethi) [reply .gitignore:20]: removing, have a look at other comments and reply, will fix shortly right after

**Comment by hiteshsethi** [comment]
@hsluoyz any changes left?

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
lgtm

**Pull request merged**